File: app/src/main/java/com/example/savourit/adapters/FriendsAdapter.java
package com.example.savourit.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.savourit.R;
import com.example.savourit.models.Friend;

import java.util.List;

public class FriendsAdapter extends RecyclerView.Adapter<FriendsAdapter.FriendViewHolder> {

    //Linru Wang
    private List<Friend> friendsList;
    private OnFriendClickListener onFriendClickListener;
    private OnFriendLongClickListener onFriendLongClickListener;

    public interface OnFriendClickListener {
        void onFriendClick(Friend friend);
    }

    public interface OnFriendLongClickListener {
        void onFriendLongClick(Friend friend);
    }

    public FriendsAdapter(List<Friend> friendsList, OnFriendClickListener listener, OnFriendLongClickListener longClickListener) {
        this.friendsList = friendsList;
        this.onFriendClickListener = listener;
        this.onFriendLongClickListener = longClickListener;
    }

    public static class FriendViewHolder extends RecyclerView.ViewHolder {
        TextView usernameTextView;

        public FriendViewHolder(View itemView) {
            super(itemView);
            usernameTextView = itemView.findViewById(R.id.txtUsername);
        }
    }

    @NonNull
    @Override
    public FriendViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_friend, parent, false);
        return new FriendViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull FriendViewHolder holder, int position) {
        Friend friend = friendsList.get(position);
        holder.usernameTextView.setText(friend.getUsername());

        //Click (for chat, view, delete)
        holder.usernameTextView.setOnClickListener(view -> {
            if (onFriendClickListener != null) {
                onFriendClickListener.onFriendClick(friend);
            }
        });

        // Long Click (Right-click equivalent) - Now handled safely
        holder.usernameTextView.setOnLongClickListener(view -> {
            onFriendLongClickListener.onFriendLongClick(friend);
            return true;  // Consume the event to prevent extra clicks
        });
    }

    public void updateList(List<Friend> newList) {
        this.friendsList.clear();
        this.friendsList.addAll(newList);
        notifyDataSetChanged();
    }


    @Override
    public int getItemCount() {
        return friendsList.size();
    }
}

File: app/src/main/java/com/example/savourit/adapters/FriendsAdapter.kt
package com.example.savourit.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.savourit.R
import com.example.savourit.models.Friend

class FriendsAdapter(
    //Linru Wang
    private val friendsList: MutableList<Friend>,
    private val onFriendClick: (Friend) -> Unit  //Callback for friend click
) : RecyclerView.Adapter<FriendsAdapter.FriendViewHolder>() {

    class FriendViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val usernameTextView: TextView = itemView.findViewById(R.id.txtUsername)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FriendViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_friend, parent, false)
        return FriendViewHolder(view)
    }

    override fun onBindViewHolder(holder: FriendViewHolder, position: Int) {
        val friend = friendsList[position]
        holder.usernameTextView.text = friend.username

        //Show menu when clicking a friend
        holder.usernameTextView.setOnClickListener {
            onFriendClick(friend)
        }
    }

    override fun getItemCount(): Int = friendsList.size
}
